### ISSUE TITLE
Is it possible to see server side http requests that take place within ajax calls?

### ISSUE THREAD

**Issue by scottfennell-toptal** [johnbillion/query-monitor#745]
Hello, I have the following setup...
- Some JS makes a fetch request to my WP API endpoint.
- The endpoint itself makes an http request to stripe.

To complicate matters, the http request to stripe is via their official php sdk, which obviously does not use wp_remote_*() and I'm treating as a black box I don't want to tamper with.

In any event, I don't see the server side http calls in query monitor.  Should I expect to?  Is there a way to?  Some of my stripe calls seem to be taking a long time and I'd like to do some performance debugging.

**Comment by crstauf** [comment]
Definitely should not expect to see those requests in QM.

You could use QM's profiling feature around the request, but that will only show time and memory.

**Comment by scottfennell-toptal** [comment]
Thank you for the response!

Can you elaborate on what it is about those request that QM is unable to see?

- Is it because they take place within the context of an ajax to a server side proxy?
- Is it because they do not use wp_remote_*()?
- Is it because they use the stripe sdk?
- Some combo?
- Something else?

I am interested in profiling them for sure.  Could you point me to any docs for that?

**Comment by crstauf** [comment]
It is because the requests are made without using WP helper functions (ex: `wp_remote_get()`).

If you review [the collector for the HTTP](https://github.com/johnbillion/query-monitor/blob/develop/collectors/http.php) data, you'll see it connects to several WordPress hooks to watch for and gather data on HTTP requests.

**Comment by scottfennell-toptal** [comment]
Now I'm tempted to tamper with the stripe sdk and make it use wp_remote_() just so I can use QM 🤣

**Comment by crstauf** [comment]
You certainly could, but if you updated Stripe's PHP SDK version, you'd have to do it over again. It's probably more pain than it's worth, but that's ultimately your judgement.

**Comment by scottfennell-toptal** [comment]
oh for sure, I just meant on staging to do some performance debugging here.

**Comment by lkraav** [comment]
Most SDK use well-known libs themselves, such as Guzzle. Could identify the lib, and write a simple QM integration plugin for it.

Although https://github.com/stripe/stripe-php/blob/master/composer.json seems to indicate they're using `curl` directly.

**Comment by johnbillion** [comment]
Over the years I've spent quite a bit of time investigating whether QM could automatically detect HTTP requests made with Guzzle or directly with Curl without having access to a variable that holds the instance or the handle. It's just not possible without using PHP ticks, which are way too resource intensive to use.

I have thought about writing a QM connector for Guzzle and Curl but the majority of the time it would still rely on a) a user realising that a plugin is doing such requests, and then b) some sort of modification to the plugin to expose its Guzzle instance or Curl handle.

**Comment by tomjn** [comment]
@johnbillion I had an idea for this, would it be possible to add a filter we could pass a Guzzle object through or a QM function to retrieve a requests object that can log the data? I know it's not automatic but in many cases we're in control of the guzzle setup but don't know how to expose it to QM. It's not as elegant as automatic pickup of guzzle which I can't see happening for the same reasons you gave

**Comment by johnbillion** [comment]
Yeah I'd be happy to add some middleware to QM that can be attached to a Guzzle request
